Tests/SonarScanner.MSBuild.Tasks.UnitTests/GetAnalyzerSettingsTests.cs

```diff
@@ -305,144 +273,140 @@ public void GetAnalyzerSettings_ConfigExists_NewBehaviour_SettingsMerged()
+        [DataRow("7.4", "vbnet", @"c:\vbnet-normal.ruleset", /* not set */ null)]
+        public void GetAnalyzerSettings_ConfigExists_TestProject_SonarAnalyzerSettingsUsed(string sonarQubeVersion, string language, string expectedRuleset, string excludeTestProject)
         {
             // Arrange and Act
-            var executedTask = ExecuteGetAnalyzerSettings_ConfigExists_TestProject_SonarAnalyzerSettingsUsed("7.3", "vbnet");
+            var executedTask = ExecuteGetAnalyzerSettings_ConfigExists_TestProject_SonarAnalyzerSettingsUsed(sonarQubeVersion, language, true, excludeTestProject);
 
             // Assert
-            executedTask.RuleSetFilePath.Should().Be("c:\\test.project.ruleset.vb");
-            executedTask.AnalyzerFilePaths.Should().BeEquivalentTo("c:\\sonar.vbnet1.dll", "c:\\Google.Protobuf.dll");
-            executedTask.AdditionalFilePaths.Should().BeEquivalentTo("c:\\add1.txt.vb", "d:\\replaced1.txt");
+            executedTask.RuleSetFilePath.Should().Be(expectedRuleset);
+            executedTask.AnalyzerFilePaths.Should().BeEquivalentTo(@"c:\wintellect1.dll", @"c:\Google.Protobuf.dll", $@"c:\sonar.{language}.dll", @"c:\Google.Protobuf.dll");
+            executedTask.AdditionalFilePaths.Should().BeEquivalentTo($@"c:\add1.{language}.txt", @"d:\replaced1.txt", "original.should.be.removed.for.excluded.test.txt"); // This test is not excluded
         }
 
-        [TestMethod]
-        public void GetAnalyzerSettings_ConfigExists_NewBehaviour_TestProject_SonarAnalyzerSettingsUsed_VB()
+        [DataTestMethod]
+        [DataRow("7.3", "cs", @"c:\csharp-deactivated.ruleset", "true", DisplayName = "Legacy CS")]
+        [DataRow("7.4", "cs", @"c:\csharp-deactivated.ruleset", "TRUE")]
+        [DataRow("7.3", "vbnet", @"c:\vbnet-deactivated.ruleset", "True", DisplayName = "Legacy VB")]
+        [DataRow("7.4", "vbnet", @"c:\vbnet-deactivated.ruleset", "tRUE")]
+        public void GetAnalyzerSettings_ConfigExists_ExcludedTestProject_DeactivatedSonarAnalyzerSettingsUsed(string sonarQubeVersion, string language, string expectedRuleset, string excludeTestProject)
         {
             // Arrange and Act
-            var executedTask = ExecuteGetAnalyzerSettings_ConfigExists_TestProject_SonarAnalyzerSettingsUsed("7.4", "vbnet");
+            var executedTask = ExecuteGetAnalyzerSettings_ConfigExists_TestProject_SonarAnalyzerSettingsUsed(sonarQubeVersion, language, true, excludeTestProject);
 
             // Assert
-            executedTask.RuleSetFilePath.Should().Be("c:\\test.project.ruleset.vb");
-            executedTask.AnalyzerFilePaths.Should().BeEquivalentTo("c:\\sonar.vbnet1.dll", "c:\\Google.Protobuf.dll");
-            executedTask.AdditionalFilePaths.Should().BeEquivalentTo("c:\\add1.txt.vb", "d:\\replaced1.txt");
+            executedTask.RuleSetFilePath.Should().Be(expectedRuleset);
+            executedTask.AnalyzerFilePaths.Should().BeEquivalentTo($@"c:\sonar.{language}.dll", @"c:\Google.Protobuf.dll");
```

Comment:
[learn] why is the `wintellect1.dll` missing here?

Reply:
SQ<=8.8: No analysis (except metrics) is performed for test projects. No SonarAnalyzer and no 3rd party analyzers (like `wintellect`) as well.
Current state: Preserves the same behavior when tests are excluded.

Reply:
is this behavior documented somewhere?
This should be documented both internally, in code comments on the Task itself, and externally in product documentation.
Maybe open a ticket to update docs (or update an existing ticket to not forget to mention this)?

Reply:
Note added in #976